src/common/k8s-api/api-manager.ts

```diff
@@ -27,7 +27,11 @@ import type { KubeApi } from "./kube-api";
 import type { KubeObject } from "./kube-object";
 import { IKubeObjectRef, parseKubeApi, createKubeApiURL } from "./kube-api-parse";
 
-export class ApiManager {
+export interface IGettableStore {
```

Comment:
Personally I really dislike the `I` prefixing of interfaces. 

Is it also possible to have better names for these interfaces, currently they seem too generic.

Reply:
On `I`-prefixing: Together with `T`-prefixing, this has many times proven valuable tool in making some more difficult typings understandable. Also it would be good to have the code somehow look off, when a coder is supposed to use an interface instead of concretion. With this, we have a better chance of making *Inversion of Control* and *loose coupling* happen on team level :)

On interface naming: Agreed. New interfaces added to better communicate and document intent.

Reply:
I disagree, would you please provide some source for your claim? I find them distracting. Items should be obvious what they are from how they are used.

Reply:
Which claim? :) A general goal here is to make stuff more obvious. Lack of Inversion of Control makes code less obvious by being a tangle of concrete dependencies. I would say that anything that could help in fixing this is a welcome thing to be tried.

Reply:
This claim:

> On I-prefixing: Together with T-prefixing, this has many times proven valuable tool in making some more difficult typings understandable.

Reply:
I'd go with the commonly used convention in TypeScript projects, e.g. not using the `I`-prefix.

Reply:
I tend to agree. At least have only one way throughout the codebase. Either prefix all or nothing.

Reply:
@panuhorsmalahti let's not make a vote out this, as best rationale should win. What is yours? Convention for sake of convention does not count :)

Reply:
I have found that not having such prefixes limits the space of possible names so that name collisions happens more often within badly designed code.

Reply:
For code style questions commonly used conventions do matter, but are not the only factor. Following style that is used in, say, 90% of projects, makes it easier for new developers to jump in. Using something else requires extra arguments (that have been provided here). In any case, my own opinion is that the `I`-prefix makes the code less aesthetic and makes the code slightly more awkward to read and type. Prefixes do have a (minor) cost, otherwise we could add them to every single thing freely.

Reply:
@Nokel81 
1. For example, we use T and I as key pieces if information when talking difficult subjects with other coders.

2. Also there was my other point:
> Also it would be good to have the code somehow look off, when a coder is supposed to use an interface instead of concretion. With this, we have a better chance of making Inversion of Control and loose coupling happen on team level :)

3. Here's a particularly dodgy piece of typing which debatably benefits from prefixes more than not:
```javascript
export interface IDependencyInjectionContainer {
    inject: <
      TInjectable extends IInjectable<TInstance, TDependencies>,
      TInstance,
      TDependencies,
    >(
      injectableKey: TInjectable,
    ) => ReturnType<TInjectable["instantiate"]>;
  }
```

Reply:
I think that we have common goal here: be able to separate interfaces from actual implementations. 

That being said, for me, it doesn't matter what is the solution to achieve that, but I haven't heard any better alternatives than prefixing them with something. Interfaces and implementations are separate for a reason, so how do you think we should separate them?

Reply:
We should separate them by name, but we don't need to "namespace" the interfaces with prefixes.

Reply:
@Nokel81 Could you provide concrete example? :)

Reply:
@Nokel81 but what about the 3 arguments? :)

Reply:
@panuhorsmalahti I agree, yet still find the other arguments more pressing.

After all, we are trying to make a big thing happen here, with Inversion of Control and all. Expect more eggs to be broken :)

Reply:
1. When taking it is quite easy to say "an interface" or "a class/implementation" called "Foo", when working in an IDE, there is hover intellisense and "goto type definition" and such.
2. I would need to an example as to why prefixing helps. The general rule of "consume interfaces, provide implementations" doesn't need any naming conventions to be realized. 
3. Because of how typescript type parameters work all those prefixes are unnecessary, `T extends F` means that `F` must be some sort of interface, and the fact that `T` is used as a parameter type means that it must be an implementation.

Reply:
@jweak we covered your point IRL. Nice to meet you! :)

Reply:
@Nokel81

> When taking it is quite easy to say "an interface" or "a class/implementation" called "Foo"

That's too much overhead to be practical.

> when working in an IDE, there is hover intellisense and "goto type definition" and such.

One shouldn't be required to hover stuff to make code easier to read in this way.

> The general rule of "consume interfaces, provide implementations" doesn't need any naming conventions to be realized.

The good rule shouldn't need the conventions, but it sure does benefit from them, as transgressions get easier to spot.

> Because of how typescript type parameters work all those prefixes are unnecessary, `T extends F` means that `F` must be some sort of interface, and the fact that `T` is used as a parameter type means that it must be an implementation.

This comes with additional cognitive load.

Reply:
> That's too much overhead to be practical.

I disagree, saying the prefix is more cumbersome as it won't ever flow with the rest of the name

> One shouldn't be required to hover stuff to make code easier to read in this way.

Why not, having too much information at a time can make code hard to read so having an optional way of getting more information when necessary is very useful.

> The good rule shouldn't need the conventions, but it sure does benefit from them, as transgressions get easier to spot.

Maybe but that doesn't rule out the fact that this convention in particular (adding predefined prefixes and/or suffixes) is an eyesore. Beautiful code should be readable and understandable without this crutch.

This convention naturally leads to doing what C# and Java do with `FooImpl implements IFoo` which just is terrible.

> This comes with additional cognitive load, making it a losing bargain.

How is more cognitive load then the convention? The code reader still has to understand what is going on

Reply:
Most of the existing interfaces are without `I` prefix and we cannot really change those without breaking extension api types. So my recommendation is to use that convention for now. IMHO main thing that matters here is to have a convention.

Reply:
@jakolehm That is a pressing argument! I might know some work-arounds, but I don't have enough visibility at this time.

Moving on! :)